docs/GETTING_STARTED.md

```diff
@@ -0,0 +1,281 @@
+`docker.io/library/rabbitmq:latest`.
+
+To do this with ctr use the following command:
+```
+ctr i pull docker.io/library/rabbitmq:latest
+```
+To confirm this image (or the image you have chosen) is in containerd's local
+content store please run:
+```
+ctr i ls
+```
+and you should see the image.  Now we need to add an image reference it so that
+it can be added to our repository.
+```
+ctr i tag docker.io/library/rabbitmq:latest localhost:5000/rabbitmq:latest
+```
+Note that the `localhost:5000` is the address of our registry, `rabbitmq` will
+be the name of the repository, and `latest` will be the tag within the
+repository.
+
+To confirm this worked once more run `ctr i ls` and see that there is now a new
+line with the tag you just created.  Also confirm that the digest, the long
+alphanumeric hash that is prefixed by "sha256:", matches the base image.
+
+Finally, we can push to the local registry:
+
+```
+ctr i push --platform linux/amd64 --plain-http localhost:5000/rabbitmq:latest
+```
+
+**NOTE**: the platform tag might be different depending what environment you
+are working in.
+
+Once again let's confirm via a curl command:
+```
+curl http://localhost:5000/v2/_catalog
+```
+We should now see that our repository has been created
+`{"repositories":["rabbitmq"]}`.  We can further inspect and see what tags were
+created (you may need to change 'rabbitmq' if you named your repo something else).
+```
+curl http://localhost:5000/v2/rabbitmq/tags/list
+```
+We can also look at the details of the tag using (you may need to change
+'rabbitmq' and 'latest' if you named your repo and tag something else):
+```
+curl http://localhost:5000/v2/rabbitmq/manifests/latest
+```
+## Create and push a SOCI index
+Now that we have confidence that our registry contains the image let's work on
+the SOCI index.  This is the pre-built index that allows random access of the
+compressed layer. For more detail please see [our
+README](https://github.com/awslabs/soci-snapshotter#no-image-conversion.)
+
+First off, we need to build the SOCI project using the `make` command.  This
+will generate two binaries in the `out` directory.  You should see:
+- soci (this is the CLI tool)
+- soci-snapshotter-grpc (this is the daemon used for lazy loading, we will use
+this later)
+
+Many of the `soci` CLI tool commands will need to be run as sudo, this is
+because the metadata is saved in directories that a non-root user often
+does not have access to.
+
+### Creating the SOCI index
+
+To create the index run:
+```
+sudo ./soci create localhost:5000/rabbitmq:latest
+```
+Behind the scenes SOCI is interacting with containerd.  So the tag
+`localhost:5000/rabbitmq` has to exist in containerd - that is what SOCI is
+referencing.  This created two kinds of objects.  The first is a series of ztocs
+(one per layer).  A ztoc is a table of contents for compressed data (z is the
+compression, toc is Table Of Contents).  The other is a manifest that relates
```

Comment:
This is confusing because the output of `soci create` just shows the layers. I went to the output below expecting to see a `manifest` nestled in amongst the `layer`s. But, of course, I didn't. Then later, `f5f2a855` just seems to show up out of thin air.

Maybe it *should* output the manifest digest.